### PR TITLE
config rework for elasticsearch upgrade

### PR DESCRIPTION
**Pull request by pgvishnuram**
## Description

This PR reconfigures elastic search to use new roles changes and adapts to 8.x 

Note: **Do not Merge this PR until the works is verified and validated**

## Related Issues

https://github.com/astronomer/issues/issues/4296

https://github.com/astronomer/issues/issues/4160

## Testing

QA should able to validate node roles using steps added in the issue ticket

## Merging

cherry-pick to release-0.30,0.32

### CODE REVIEW

File: charts/elasticsearch/templates/es-configmap.yaml [hunk 1]
```diff
@@ -19,11 +19,6 @@ data:
         - {{ $releaseName }}-elasticsearch-master-{{ $i }}
         {{- end }}
     node:
-      master: ${NODE_MASTER}
-      name: ${NODE_NAME}
-      data: ${NODE_DATA}
-      ingest: ${NODE_INGEST}
-      max_local_storage_nodes: ${MAX_LOCAL_STORAGE_NODES}
```
Comment (pgvishnuram) [inline charts/elasticsearch/templates/es-configmap.yaml:26]: these values are deprecated so removing these values from our actual config

File: charts/kibana/templates/ingress.yaml [hunk 2]
```diff
@@ -21,6 +21,8 @@ metadata:
     nginx.ingress.kubernetes.io/auth-url: https://houston.{{ .Values.global.baseDomain }}/v1/authorization
     nginx.ingress.kubernetes.io/auth-signin: https://app.{{ .Values.global.baseDomain }}/login
     nginx.ingress.kubernetes.io/auth-response-headers: authorization, username, email
+    nginx.ingress.kubernetes.io/configuration-snippet: |
+      more_set_headers "Content-Security-Policy: script-src 'self'";
```
Comment (pgvishnuram) [inline charts/kibana/templates/ingress.yaml:25]: this config snipet is required by kibana after migrating to 8.x due to strict requirement in CSP rules

File: charts/elasticsearch/values.yaml [hunk 3]
```diff
@@ -119,6 +124,17 @@ data:
...
+    - data
+    - data_cold
+    - data_content
+    - data_frozen
+    - data_hot
+    - data_warm
+    - ml
+    - remote_cluster_client
+    - transform
+
```
Comment (pgvishnuram) [inline charts/elasticsearch/values.yaml:137]: there are default roles added from our existing config

File: charts/elasticsearch/values.yaml [hunk 4]
```diff
@@ -154,6 +170,11 @@ master:
     NODE_INGEST: "false"
     HTTP_ENABLE: "false"
 
+  roles:
+    - master
+    - ml
+    - remote_cluster_client
```
Comment (pgvishnuram) [inline charts/elasticsearch/values.yaml:176]: there are default roles added from our existing config for master service

Comment (danielhoherd) [reply charts/elasticsearch/values.yaml:176]: Yeah, I see those listed in the error message shown in this ticket https://github.com/astronomer/issues/issues/4160

File: charts/elasticsearch/templates/_helpers.tpl [hunk 5]
```diff
@@ -136,3 +136,22 @@ imagePullSecrets:
   - name: {{ .Values.global.privateRegistry.secretName }}
 {{- end -}}
 {{- end -}}
+
+
```
Comment (danielhoherd) [inline charts/elasticsearch/templates/_helpers.tpl:140]:
```suggestion

```

File: charts/elasticsearch/templates/es-configmap.yaml [hunk 6]
```diff
@@ -54,7 +49,12 @@ data:
         - {{ template "elasticsearch.fullname" . }}-headless-discovery
 
     # Disable scripting for security.
+    {{ if semverCompare "<8" .Values.images.es.tag -}}
     script.allowed_types: none
+    {{- else -}}
+    script.allowed_types: inline
```
[the remark this answers is not in the document]
Comment (pgvishnuram) [reply charts/elasticsearch/templates/es-configmap.yaml:55]: https://www.elastic.co/guide/en/elasticsearch/reference/current/modules-scripting-security.html#allowed-script-types-setting - kibana needs them

[the remark this answers is not in the document]
Comment (pgvishnuram) [reply charts/elasticsearch/templates/es-configmap.yaml:55]: kibana needs them and with upgrade assistant its like an hard requirement

[the remark this answers is not in the document]
Comment (danielhoherd) [reply charts/elasticsearch/templates/es-configmap.yaml:55]: ok, sounds like we don't have a choice. that's unfortunate.

[the remark this answers is not in the document]
Comment (danielhoherd) [reply charts/elasticsearch/templates/es-configmap.yaml:55]: Can we just remove the conditional block for <8 ? We're not going to support it, so I don't see a good reason to leave it in.

**Review by danielhoherd — CHANGES_REQUESTED** [review]
Code looks good. Needs tests.

**Comment by danielhoherd** [comment]
@pgvishnuram we need to bump to curator 8.0.4 in this PR since that is required in order for the changes to be fully compatible. That image is already available in in the repo <https://quay.io/repository/astronomer/ap-curator?tab=tags>

**Review by rishkarajgi — APPROVED** [review]

**Pull request merged**